stdio/scanf.c

```diff
@@ -510,8 +528,58 @@ static int scanf_parse(char *ccltab, const char *inp, char const *fmt0, va_list
 				nread += p - buf;
 				nconversions++;
 				break;
-			}
+
+			case CT_FLOAT:
+				if (width == 0 || width > sizeof(buf) - 1) {
+					width = sizeof(buf) - 1;
+				}
+
+				if (strtold(inp, &p) == 0 && (inp == p || errno == ERANGE)) {
+					return (nconversions != 0 ? nassigned : -1);
+				}
+
+				if ((size_t)(p - inp) >= width) {
+					return (nconversions != 0 ? nassigned : -1);
+				}
+
+				width = p - inp;
+
+				p = buf;
+				while (width > 0) {
+					if (inr <= 0) {
+						return (nconversions != 0 ? nassigned : -1);
+					}
+					*p++ = *inp++;
```

Comment:
A suggestion, so the intend is more clear without checking operator precedence table
```suggestion
					*(p++) = *(inp++);
```

Reply:
You're right. I've duplicated this fragment from the working `CT_INT` case above, in the current scanf implementation, so maybe let's leave it as it is for now, since it's a quick-fix that has recently been verified, let's merge it. I found a lot of MISRA compatibility issues during my scanf rework, so I will address them in a separate PR (also for suggested change).

Reply:
Ack